src/lib/model_selection/_split.ts

```diff
@@ -208,3 +210,188 @@ export function train_test_split(
     yTrain: clean(yTrain),
   };
 }
+
+const rangeValidationError = (type, size, n_samples) => `${type}=${size} should be either 
+positive and smaller than number of samples ${n_samples} or a float in (0, 1) range`;
+
+const testRangeValidationError = (test_size, n_samples) => rangeValidationError('test_size', test_size, n_samples);
+
+const trainRangeValidationError = (test_size, n_samples) => rangeValidationError('test_size', test_size, n_samples);
+
+function uniq(array: Type1DMatrix<any> = null): Type2DMatrix<number> {
+  const uniqArray = _.uniq(_.flatten(array)).sort();
+  let min = Number.MAX_VALUE;
+  let max = Number.MIN_VALUE;
+
+  const valueCount = {};
+  const uniqIndexMap = uniqArray.reduce((acc, ele, i) => {
+    if (min > ele) {
+      min = ele;
+    }
+
+    if (max < ele) {
+      max = ele;
+    }
+
+    return {
+      ...acc,
+      [ele]: i,
+    };
+  }, {});
+
+  const indexMap = array.map((ele) => {
+    if (valueCount[ele]) {
+      valueCount[ele] += 1;
+    } else {
+      valueCount[ele] = 1;
+    }
+    return uniqIndexMap[ele];
+  });
+
+  return [uniqArray, indexMap];
+}
+
+function binCount(array: number[]) {
```

Comment:
Same here, could you think of a way to use tfjs API instead of raw JS API?

Reply:
Also, do you see this function being used elsewhere? If you think it will be used for more than StartifiedShuffleSplit, please move this to utils/tensors